ipa-core/src/net/test.rs

```diff
@@ -318,17 +508,40 @@ impl TestServerBuilder {
     }
 }
 
-fn get_test_certificate_and_key(id: HelperIdentity) -> (&'static [u8], &'static [u8]) {
+pub struct ClientIdentities {
+    pub helper: ClientIdentity<Helper>,
+    pub shard: ClientIdentity<Shard>,
+}
+
+#[must_use]
+pub fn create_ids(disable_https: bool, id: HelperIdentity, ix: ShardIndex) -> ClientIdentities {
+    if disable_https {
+        ClientIdentities {
+            helper: ClientIdentity::Header(id),
+            shard: ClientIdentity::Header(ix),
+        }
+    } else {
+        get_client_test_identity(id, ix)
+    }
+}
+
+fn get_test_certificate_and_key(id: usize) -> (&'static [u8], &'static [u8]) {
     (TEST_CERTS[id], TEST_KEYS[id])
 }
 
+/// Creating a cert client identity. Using the same certificate for both shard and mpc.
 #[must_use]
-pub fn get_test_identity(id: HelperIdentity) -> ClientIdentity {
-    let (mut certificate, mut private_key) = get_test_certificate_and_key(id);
-    ClientIdentity::from_pkcs8(&mut certificate, &mut private_key).unwrap()
+pub fn get_client_test_identity(id: HelperIdentity, ix: ShardIndex) -> ClientIdentities {
+    let pos = ix.as_index() * 3 + id.as_index();
```

Comment:
that deserves a comment and a common function to compute the absolute offset as I believe it is used in more than one place

Reply:
Ok